xds/src/main/java/io/grpc/xds/XdsRoutingLoadBalancer.java

```diff
@@ -16,18 +16,209 @@
 
 package io.grpc.xds;
 
+import static io.grpc.ConnectivityState.CONNECTING;
+import static io.grpc.ConnectivityState.IDLE;
+import static io.grpc.ConnectivityState.READY;
+import static io.grpc.ConnectivityState.TRANSIENT_FAILURE;
+import static io.grpc.xds.XdsSubchannelPickers.BUFFER_PICKER;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.grpc.ConnectivityState;
+import io.grpc.InternalLogId;
 import io.grpc.LoadBalancer;
+import io.grpc.MethodDescriptor;
 import io.grpc.Status;
+import io.grpc.internal.ServiceConfigUtil.PolicySelection;
+import io.grpc.util.ForwardingLoadBalancerHelper;
+import io.grpc.util.GracefulSwitchLoadBalancer;
+import io.grpc.xds.XdsLogger.XdsLogLevel;
+import io.grpc.xds.XdsRoutingLoadBalancerProvider.MethodName;
+import io.grpc.xds.XdsRoutingLoadBalancerProvider.Route;
+import io.grpc.xds.XdsRoutingLoadBalancerProvider.XdsRoutingConfig;
+import io.grpc.xds.XdsSubchannelPickers.ErrorPicker;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
 
-// TODO(zdapeng): Implementation.
 /** Load balancer for xds_routing policy. */
 final class XdsRoutingLoadBalancer extends LoadBalancer {
 
+  private final XdsLogger logger;
+  private final Helper helper;
+  private final Map<String, GracefulSwitchLoadBalancer> routeBalancers = new HashMap<>();
+  private final Map<String, RouteHelper> routeHelpers = new HashMap<>();
+
+  private Map<String, PolicySelection> actions = ImmutableMap.of();
+  private List<Route> routes = ImmutableList.of();
+
+  XdsRoutingLoadBalancer(Helper helper) {
+    this.helper = helper;
```

Comment:
nit: keep our convention `checkNotNull`, although the instantiation is completely internal. Maybe it's always a good practice to check null in constructors for arguments that are saved as fields and to be used later.

Reply:
done.